File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int n = digits.length;
        if(digits[n-1]!=9){
            digits[n-1]+=1;
            return digits;
        }
        int[] ans = new int[n+1];
        int i=n-1;
        while(i>=0 && digits[i]==9){
            ans[i+1]=0;
            digits[i]=0;
            i--;
        }
        if(i==-1){
            ans[0] = 1;
            return ans;
        }
        if(i>=0){
            // ans[i] = digits[i]+1;
            digits[i]+=1;
            i--;
        }
        // while(i>=0){
        //     ans[i] = digits[i]
        // }
        return digits;
    }
}

File: 0066-plus-one/0066-plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        if(digits[digits.size()-1]==9){
            int i = digits.size()-1;
            while(i>=0 && digits[i]==9){
                digits[i]=0;
                i--;
            }
            if(i==-1){
                digits[0]=1;
                digits.push_back(0);
            }
            else{
                digits[i]++;
            }
            return digits;
        }
        else{
            digits[digits.size()-1]++;
            return digits;
        }
    }
};